src/Cardano/WalletLayer/Wallet.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE TypeOperators #-}
+
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- | Module provides internal implementation of
+-- | wallet api
+
+module Cardano.WalletLayer.Wallet
+    (
+      CreateWallet
+    , WalletId
+    ) where
+
+import Prelude
+
+import Cardano.Wallet
+    ( Wallet )
+import Cardano.Wallet.AddressDerivation
+    ( Passphrase )
+import Cardano.Wallet.AddressDiscovery
+    ( AddressPoolGap )
+import Cardano.Wallet.Mnemonic
+    ( Mnemonic )
+import Data.Text
+    ( Text )
+import Data.Time.Units
+    ( Microsecond )
+import GHC.Generics
+    ( Generic )
+
+-- | Errors
+data CreateWalletError
+
+data GetWalletError
+
+-- | Types
+data CreateWallet = CreateWallet NewWallet | ImportWallet
+
+data NewWallet = NewWallet
+    {
+      _mnemonicSentence
+        :: !(Mnemonic 15)
+    , _mnemonicSentencePassphrase
+        :: !(Maybe (Passphrase "generation"))
+    , _name
+        :: !WalletName
+    , _passphrase
+        :: !(Passphrase "encryption")
+    , _addressPoolGap
+        :: !AddressPoolGap
+    } deriving (Show, Generic)
+
+newtype WalletName = WalletName { _getWalletName :: Text }
+    deriving (Eq, Show)
```

Comment:
Let's not prefix names with `_`, if we end up using lenses through `generic-lens`, the record fields name will be our lenses, so we can just use them as is :rainbow:

Reply:
removed `_`